cpu/sam0_common/periph/i2c.c

```diff
@@ -171,6 +181,29 @@ void i2c_release(i2c_t dev)
     mutex_unlock(&locks[dev]);
 }
 
+void i2c_deinit(i2c_t dev)
+{
+    assert(dev < I2C_NUMOF);
+
+    mutex_lock(&locks[dev]);
+    _i2c_poweroff(dev);
+
+    gpio_init_mux(i2c_config[dev].sda_pin, GPIO_MUX_A);
+    gpio_init_mux(i2c_config[dev].scl_pin, GPIO_MUX_A);
```

Comment:
There is no need to switch them to GPIO_MUX_A.

Reply:
This should mux the GPIO pin back to default (i.e. GPIO_MUX_A). Thus, the GPIO driver gets back control of these pins.

Reply:
But GPIO_MUX_A is the EIC peripheral not PORT. If you want to get back to gpio driver you just have to clear bit PMUXEN from PINCFG.
PMUX register has nothing to do with this business, this is why gpio_init_mux shouldn't be call here.

Reply:
You are right, I've added `gpio_disable_mux()` to `periph/gpio.c`.